### PR TITLE
Checking for weak signatures

### PR DESCRIPTION
**Pull request by Diagoras**

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
Fix: #11

File: src/main/java/com/github/s4u/plugins/PGPVerifyMojo.java [hunk 1]
```diff
@@ -312,6 +321,15 @@ private void initCache() throws MojoFailureException {
 
     private boolean verifyPGPSignature(Artifact artifact, File artifactFile, File signatureFile) throws MojoFailureException {
 
+        final Map<Integer, String> weakSignatures = ImmutableMap.<Integer, String>builder()
+                .put(1, "MD5")
+                .put(4, "DOUBLE_SHA")
+                .put(5, "MD2")
+                .put(6, "TIGER_192")
+                .put(7, "HAVAL_5_160")
+                .put(11, "SHA224")
```
Comment (netmackan) [inline src/main/java/com/github/s4u/plugins/PGPVerifyMojo.java]: Is it intended that SHA-224 is considered weak but not SHA-1?

Comment (Diagoras) [inline src/main/java/com/github/s4u/plugins/PGPVerifyMojo.java]: Yes, although I know it's weird. My reasoning is that SHA-1 is so commonly used that marking it as weak at this point would flood developers with warning/error messages, even though it's strictly weaker than SHA-224. Essentially the same reasoning used by the developer of OpenKeychain [when they built their list of allowed signature algorithms](https://github.com/open-keychain/open-keychain/blob/master/OpenKeychain/src/main/java/org/sufficientlysecure/keychain/pgp/PgpSecurityConstants.java).

I originally checked the security of the underlying public key as well, but the same problem occurred (so, so many RSA keys under 2048 bits...). One of the chief advantages of this plugin (IMO) is its SSH-like security model - as a rule, it won't spam warnings at you unless something is seriously wrong and it won't fail the build unless you're probably under attack (at least under default settings), so I'm hesitant to add a whole bunch of error messages for common cases that developers can't do anything about.

That being said, if we judge that SHA224 is still an improvement over SHA1 (which it is by bits of security, if not frequency of use) then I can remove it from this list.

Comment (netmackan) [inline src/main/java/com/github/s4u/plugins/PGPVerifyMojo.java]: Ok, thanks. Just noticed it and thought it looked strange but now I know it is a reason behind it.

**Pull request merged**